src/components/SearchContainer/SearchContainer.spec.js

```diff
@@ -0,0 +1,41 @@
+import SearchContainer from './SearchContainer';
+import { MemoryRouter } from 'react-router';
+import Router from 'react-router-dom';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn(),
+  useLocation: jest.fn(),
+}));
+
+describe('Search testing', () => {
+  jest.spyOn(Router, 'useLocation').mockReturnValue({ search: '' });
```

Comment:
The better way to test the component where router is involved is to re-create a simplified version of application routes and provide them to the `MemoryRouter` (almost like it is done in the second test) - that way you don't have to mock any of Router hooks.

Reply:
Yes I know, I wanted to try with different approaches